Index: git/Make.config
===================================================================
--- git.orig/Make.config
+++ git/Make.config
@@ -3,9 +3,14 @@
 
 ### The C compiler and options:
 
-CC       = gcc
+#CC       = gcc
 CFLAGS   = -O2
 
+# no CC given at all: force gcc
+ifeq (x$(CC)x,xx)
+  CC=gcc
+endif
+
 #CXX      = g++
 
 CXXFLAGS = -g -O2 -Wall -Woverloaded-virtual
@@ -40,10 +45,10 @@ ifndef $(DESTDIR)
 DESTDIR = /usr/local
 endif
 
-BINDIR = $(DESTDIR)/bin
-LIBDIR = $(DESTDIR)/lib
-INCDIR = $(DESTDIR)/include
-MANDIR = $(DESTDIR)/man
+BINDIR = $(DESTDIR)${prefix}/bin
+LIBDIR = $(DESTDIR)${prefix}/lib
+INCDIR = $(DESTDIR)${prefix}/include
+MANDIR = $(DESTDIR)${prefix}/man
 
 ### Includes and defines
 
@@ -80,7 +85,7 @@ HAVE_DRIVER_VNCSERVER=1
 
 # uncomment this variable if you want to enable the experimental AX 206 based digital photo frame driver
 # Read DRIVER.ax206dpf before use!
-#HAVE_DRIVER_AX206DPF=1
+HAVE_DRIVER_AX206DPF=1
 
 # uncomment this variable if you want to enable the experimental support for picoLCD 256x64 
 #HAVE_DRIVER_picoLCD_256x64=1
Index: git/Makefile
===================================================================
--- git.orig/Makefile
+++ git/Makefile
@@ -8,7 +8,7 @@ ARCHIVE = $(PROJECT)-$(VERSION)
 PACKAGE = $(ARCHIVE)
 TMPDIR = /tmp
 
-UDEVRULESDIR ?= /etc/udev/rules.d/
+UDEVRULESDIR ?= $(DESTDIR)/etc/udev/rules.d/
 UDEVRULE ?= 99-graphlcd-base.rules
 
 ### Targets:
@@ -24,7 +24,8 @@ install:
 	@$(MAKE) -C glcddrivers install
 	@$(MAKE) -C glcdskin install
 	@$(MAKE) -C tools install
-	test -d "${UDEVRULESDIR}" && install -m 644 -o root -g root "$(UDEVRULE)" "$(UDEVRULESDIR)"
+	install -d $(UDEVRULESDIR)
+	test -d "${UDEVRULESDIR}" && install -m 644 "$(UDEVRULE)" "$(UDEVRULESDIR)"
 
 uninstall:
 	@$(MAKE) -C glcdgraphics uninstall
Index: git/glcddrivers/port.c
===================================================================
--- git.orig/glcddrivers/port.c
+++ git/glcddrivers/port.c
@@ -18,7 +18,6 @@
 #include <unistd.h>
 #include <termios.h>
 #include <pthread.h>
-#include <sys/io.h>
 #include <sys/ioctl.h>
 #include <linux/ppdev.h>
 #include <linux/parport.h>
@@ -29,6 +28,7 @@
 
 #if defined(__linux__) && (defined(__i386__) || defined(__x86_64__))
   #define __HAS_DIRECTIO__ 1
+  #include <sys/io.h>
 #endif
 
 namespace GLCD
Index: git/tools/convpic/Makefile
===================================================================
--- git.orig/tools/convpic/Makefile
+++ git/tools/convpic/Makefile
@@ -36,7 +36,7 @@ $(PRGNAME): $(OBJS)
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
+	install -m 755 $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
Index: git/tools/crtfont/Makefile
===================================================================
--- git.orig/tools/crtfont/Makefile
+++ git/tools/crtfont/Makefile
@@ -35,7 +35,7 @@ $(PRGNAME): $(OBJS)
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
+	install -m 755 $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
Index: git/tools/genfont/Makefile
===================================================================
--- git.orig/tools/genfont/Makefile
+++ git/tools/genfont/Makefile
@@ -38,7 +38,7 @@ $(PRGNAME): $(OBJS)
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
+	install -m 755 $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
 	
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
Index: git/tools/lcdtestpattern/Makefile
===================================================================
--- git.orig/tools/lcdtestpattern/Makefile
+++ git/tools/lcdtestpattern/Makefile
@@ -36,7 +36,7 @@ $(PRGNAME): $(OBJS)
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
+	install -m 755 $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
Index: git/tools/showpic/Makefile
===================================================================
--- git.orig/tools/showpic/Makefile
+++ git/tools/showpic/Makefile
@@ -36,7 +36,7 @@ $(PRGNAME): $(OBJS)
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
+	install -m 755 $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
Index: git/tools/showtext/Makefile
===================================================================
--- git.orig/tools/showtext/Makefile
+++ git/tools/showtext/Makefile
@@ -36,7 +36,7 @@ $(PRGNAME): $(OBJS)
 
 install: $(PRGNAME)
 	install -d $(BINDIR)
-	install -m 755 -o root -g root $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
+	install -m 755 $(HAVE_STRIP) $(PRGNAME) $(BINDIR)
 
 uninstall:
 	rm -f $(BINDIR)/$(PRGNAME)
